File: easy/3898_find_the_degree_of_each_vertex/solution.cpp
#include <numeric>
#include <vector>

class Solution {
public:
  std::vector<int> findDegrees(std::vector<std::vector<int>> &matrix) {
    std::vector<int> degrees;
    degrees.reserve(matrix.size());

    for (const std::vector<int> &row : matrix)
      degrees.push_back(std::reduce(row.begin(), row.end()));

    return degrees;
  }
};

File: easy/3898_find_the_degree_of_each_vertex/solution.py
from typing import List


class Solution:
    def findDegrees(self, matrix: List[List[int]]) -> List[int]:
        n = len(matrix)

        degrees = [0] * n

        for i in range(n):
            for j in range(n):
                if matrix[i][j] == 1:
                    degrees[i] += 1

        return degrees
